webUI/react/src/components/reader/pager/VerticalPager.tsx

```diff
@@ -29,36 +29,87 @@ export default function VerticalReader(props: IReaderProps) {
     const classes = useStyles();
     const history = useHistory();
 
-    const [initialScroll, setInitialScroll] = useState(-1);
-    const initialPageRef = useRef<HTMLDivElement>(null);
+    const pageRef = useRef<HTMLDivElement>(null);
+    const pagesRef = useRef<HTMLDivElement[]>([]);
+    let readerCurPage = curPage;
+
+    function nextPage() {
+        if (curPage < pages.length - 1) {
+            setCurPage((x) => {
+                readerCurPage = x + 1;
+                return readerCurPage;
+            });
+            pagesRef.current[readerCurPage].scrollIntoView();
```

Comment:
the lambda function might be slow to execute and `pagesRef.current[readerCurPage].scrollIntoView();` could generate unexpected results.

Reply:
Do you mean that I should avoid using scrollIntoView() or double check that pagesRef.current[readerCurPage] is not null?

Reply:
https://reactjs.org/docs/state-and-lifecycle.html#state-updates-may-be-asynchronous so `pagesRef.current[readerCurPage].scrollIntoView();` might not have the correct `readerCurPage` at the time as it might be set in another thread.

Reply:
Also this https://reactjs.org/docs/refs-and-the-dom.html#dont-overuse-refs your use of refs makes the code nondeterministic and harder to debug, and prone to unexpected states

Reply:
Do you have any suggestions on how I can reduce my use of refs?